docs/source/write_first_app.rst

```diff
@@ -164,329 +188,673 @@ You should see the following:
+to interact with the blockchain. The section of the application code is shown below.
 
 .. code:: bash
 
-  const { Gateway, Wallets } = require('fabric-network');
+  // 2. register & enroll application user with CA, which is used as client identify to make chaincode calls, stores app user identity in local wallet
+        await registerUser.RegisterAppUser();
 
-First, the program uses the Wallet class to get our application user from our file system.
+Similar to the admin enrollment, this program uses a CSR to enroll ``appUser`` and
+store its credentials alongside those of ``admin`` in the wallet. We now have
+identities for two separate users --- ``admin`` and ``appUser`` --- that can be
+used by our application.
+
+Scrolling further down in your terminal output, you should see confirmation of the app user registration
+similar to this:
+
+.. code:: bash
+
+  Successfully registered and enrolled user appUser and imported it into the wallet
+
+Third, how the sample application interacts with the chaincode
+-----------------------------------------------------------
+
+In the prior steps, the application generated the admin and app user credentials and placed them in the wallet.
+If the credentials exist and have the correct permissions attributes associated with them, the sample application user
+will be able to call chaincode functionality from the sample application after getting reference to the channel name and contract name.
+
+You will notice that in the following lines of application code, the application is getting reference
+to the channel on the network and the contract name:
 
 .. code:: bash
 
-  const identity = await wallet.get('appUser');
+    //3. Prepare to call chaincode using fabric javascript node sdk
+    // Create a new gateway for connecting to our peer node.
+        const gateway = new Gateway();
+        await gateway.connect(ccp, {
+            wallet,
+            identity: registerUser.ApplicationUserId,
+            discovery: {enabled: true, asLocalhost: true}
+        });
+        try {
+            // Get the network (channel) our contract is deployed to.
+            const network = await gateway.getNetwork(myChannel);
 
-Once the program has an identity, it uses the Gateway class to connect to our network.
+            // Get the contract from the network.
+            const contract = network.getContract(myChaincodeName);
 
-.. code:: bash
+Fourth, initializing the ledger with some data we can work with
+---------------------------------------------------------------
 
-  const gateway = new Gateway();
-  await gateway.connect(ccpPath, { wallet, identity: 'appUser', discovery: { enabled: true, asLocalhost: true } });
+Now that we are to the point where we are actually having the sample application submit transactions, let’s
+go through them in sequence where we'll show the snippet of the application code, the snippet of the corresponding
```

Comment:
where we'll show the snippet of the application code, the snippet of the corresponding -> , look at the snippets of the application code and the corresponding

Reply:
Done.